lib/error_catalog.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <map>
+#include <string>
+#include "error_catalog.h"
+#include "error_reporter.h"
+
+std::map<int, const std::string> ErrorReporter::errorCatalog = {
```

Comment:
Why is this declared in error_reporter.h and defined in error_catalog.cpp? Circular dependency?

I hate global variables like this as you don't know when they are constructed. I'd rather see a singleton pattern here (with a method that can be called by backends to register new error codes in the catalog?)